splunk_eventgen/lib/plugins/generator/jinja.py

```diff
@@ -238,8 +239,11 @@ def gen(self, count, earliest, latest, samplename=None):
                 self.jinja_stream = jinja_loaded_template.stream(jinja_loaded_vars)
                 lines_out = []
                 try:
-                    for line in self.jinja_stream:
-                        if line != "\n":
+                    for raw_line in self.jinja_stream:
+                        # trim the newline char for jinja output
+                        # it is quite normal to output empty newlines in jinja
+                        line = raw_line.strip()
```

Comment:
we need to trim the newline and empty lines for the output of jinja template.
Because for each validate event, we assume it is a one line json string which has the format
```
{ 'time': xxx, 'raw': xxx, 'sourcetype': sss} 
```

skipping empty lines and breaking the loop is the root cause of #174

Reply:
how do I do multiline events then?  We don't want to trim "every" "\n" just the ones where the only thing in the line is "\n".

Reply:
I see the issue and get what you're trying to solve, but I don't think we can just use .strip(), gotta be another way

Reply:
To generate multiline events in the ‘raw’ field in one json formatted events, use can do this in the jinja template directly. Just like this

```{% for _ in range(0, large_number) %}
{%- time_now -%}
 {"_time":"{{ time_now_epoch }}", "_raw":"{{ time_now_formatted }}this is the first line, seq: {{loop.index}}/{{large_number}}\nthis is the second line, seq: {{loop.index}}/{{large_number}}\nthis is the thrid line, seq: {{loop.index}}/{{large_number}}"}
{% endfor %}```
the output of this jinja template is
```2019-04-24T10:47:35this is the first line, seq: 1/4
this is the second line, seq: 1/4
this is the thrid line, seq: 1/4
2019-04-24T10:47:35this is the first line, seq: 2/4
this is the second line, seq: 2/4
this is the thrid line, seq: 2/4
2019-04-24T10:47:35this is the first line, seq: 3/4
this is the second line, seq: 3/4
this is the thrid line, seq: 3/4
2019-04-24T10:47:35this is the first line, seq: 4/4
this is the second line, seq: 4/4
this is the thrid line, seq: 4/4```

Make sense?

Reply:
No worries. User can hit the enter key to make the new line directly in jinja template file. 
Jinja engine will keep what it is in your template file and render the output. No changes for newline and whitespace. 

So, this template 
```
{% for _ in range(0, large_number) %}
{%- time_now -%}
  {"_time":"{{ time_now_epoch }}", "_raw":"{{ time_now_formatted }}this is the first line, seq: {{loop.index}}/{{large_number}}
    this is the second line, seq: {{loop.index}}/{{large_number}}
    this is the thrid line, seq: {{loop.index}}/{{large_number}}"}
{% endfor %}
```
the output is
```
2019-04-24T15:58:23this is the first line, seq: 1/4
    this is the second line, seq: 1/4
    this is the thrid line, seq: 1/4
2019-04-24T15:58:23this is the first line, seq: 2/4
    this is the second line, seq: 2/4
    this is the thrid line, seq: 2/4
2019-04-24T15:58:23this is the first line, seq: 3/4
    this is the second line, seq: 3/4
    this is the thrid line, seq: 3/4
2019-04-24T15:58:23this is the first line, seq: 4/4
    this is the second line, seq: 4/4
    this is the thrid line, seq: 4/4
```